File: data/seed.js
export const seedProducts = [
  { id: "p1", name: "Premium Cat Food 2kg", price: 1450, category: "Food", stock: 42, threshold: 15, icon: "Fish", desc: "Grain-free salmon recipe for adult cats. Rich in omega-3 for a healthy coat." },
  { id: "p2", name: "Dog Chew Toy", price: 650, category: "Toys", stock: 8, threshold: 10, icon: "Bone", desc: "Durable rubber chew toy for medium to large dogs." },
  { id: "p3", name: "Fish Pellets 500g", price: 450, category: "Food", stock: 15, threshold: 20, icon: "Waves", desc: "Sinking pellets for tropical fish, boosts color & growth." },
  { id: "p4", name: "Chicken Vaccine (dose)", price: 900, category: "Medicine", stock: 0, threshold: 10, icon: "Syringe", desc: "Single-dose poultry vaccine, cold-chain verified." },
  { id: "p5", name: "Puppy Starter Food 1kg", price: 980, category: "Food", stock: 26, threshold: 10, icon: "Utensils", desc: "Balanced nutrition formulated for puppies 2–12 months." },
  { id: "p6", name: "Cat Scratching Post", price: 2200, category: "Accessories", stock: 12, threshold: 5, icon: "PawPrint", desc: "Sisal-wrapped scratching post with a plush perch on top." },
];

export const seedServices = [
  { id: "s1", name: "Health Checkup", price: 1200, duration: "30 min", icon: "Stethoscope", desc: "General wellness exam by a licensed vet." },
  { id: "s2", name: "Dog Vaccination", price: 1800, duration: "20 min", icon: "Syringe", desc: "Core vaccines for puppies & adult dogs." },
  { id: "s3", name: "Pet Grooming", price: 1500, duration: "60 min", icon: "Scissors", desc: "Bath, trim, nail clipping & de-shedding." },
  { id: "s4", name: "Dental Cleaning", price: 2200, duration: "45 min", icon: "Smile", desc: "Plaque removal & full oral health check." },
];

export const seedZones = [
  { id: "z1", name: "Zone A — Nairobi CBD", fee: 200, eta: "15–30 min" },
  { id: "z2", name: "Zone B — Westlands / Kilimani", fee: 300, eta: "30–45 min" },
  { id: "z3", name: "Zone C — Karen / Langata", fee: 450, eta: "45–60 min" },
];

export const seedOrders = [
  { id: "VT-10421", kind: "service", label: "Health Checkup", total: 1200, status: "completed", customer: "Peter Otieno", when: "2 days ago" },
  { id: "VT-10390", kind: "product", label: "3 items", total: 2150, status: "delivered", customer: "Amina Hassan", when: "4 days ago" },
];

File: backend/seed.py
"""Create a local Vetty demo database. Run: python seed.py"""
from werkzeug.security import generate_password_hash
from vetty_api import app, db, User, Product, Service, DeliveryZone, upgrade_local_schema

PRODUCTS = [
    {'name': 'Premium Cat Food 2kg', 'price': 1450, 'category': 'Food', 'stock': 42, 'threshold': 15, 'icon': 'Fish', 'description': 'Grain-free salmon recipe for adult cats.'},
    {'name': 'Dog Chew Toy', 'price': 650, 'category': 'Toys', 'stock': 8, 'threshold': 10, 'icon': 'Bone', 'description': 'Durable rubber chew toy for medium and large dogs.'},
    {'name': 'Fish Pellets 500g', 'price': 450, 'category': 'Food', 'stock': 15, 'threshold': 20, 'icon': 'Waves', 'description': 'Sinking pellets for tropical fish.'},
]
SERVICES = [
    {'name': 'Health Checkup', 'price': 1200, 'duration': '30 min', 'icon': 'Stethoscope', 'description': 'General wellness exam by a licensed vet.'},
    {'name': 'Dog Vaccination', 'price': 1800, 'duration': '20 min', 'icon': 'Syringe', 'description': 'Core vaccines for puppies and adult dogs.'},
]
ZONES = [
    {'name': 'Zone A — Nairobi CBD', 'fee': 200, 'eta': '15–30 min'},
    {'name': 'Zone B — Westlands / Kilimani', 'fee': 300, 'eta': '30–45 min'},
    {'name': 'Zone C — Karen / Langata', 'fee': 450, 'eta': '45–60 min'},
]


def seed_database():
    with app.app_context():
        db.create_all()
        upgrade_local_schema()
        if not User.query.filter_by(email='admin@vetty.co.ke').first():
            db.session.add(User(username='vetty-admin', email='admin@vetty.co.ke', role='admin', password_hash=generate_password_hash('ChangeMe123!')))
        if not Product.query.first(): db.session.add_all(Product(**item) for item in PRODUCTS)
        if not Service.query.first(): db.session.add_all(Service(**item) for item in SERVICES)
        if not DeliveryZone.query.first(): db.session.add_all(DeliveryZone(**item) for item in ZONES)
        db.session.commit()
    print('Vetty demo data ready. Change the documented admin password before deployment.')


if __name__ == '__main__':
    seed_database()
